tchannel-core/src/main/java/com/uber/tchannel/handlers/RequestRouter.java

```diff
@@ -119,6 +119,10 @@ protected void messageReceived(final ChannelHandlerContext ctx, final Request re
         // Get handler for this method
         RequestHandler handler = this.getRequestHandler(service, endpoint);
         if (handler == null) {
+            handler = topChannel.getDefaultHandler();
+        }
+
+        if (handler == null) {
```

Comment:
the default handler will be set from yarpc-java which is using tchannel-java. If user doesn't set the defaulthandler , the tchannel still does in the existing way. That's why the default handler can be null.